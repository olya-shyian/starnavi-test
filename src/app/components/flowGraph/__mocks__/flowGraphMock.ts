import { FormattedFilms } from "@/app/interfaces/IFilms";

export const heroeInfoMock: FormattedFilms[] = [
  {
    title: "A New Hope",
    id: "1",
    starships: [
      { id: "1", name: "X-Wing Starfighter" },
      { id: "2", name: "Millennium Falcon" },
    ],
  },
  {
    title: "The Empire Strikes Back",
    id: "2",
    starships: [{ id: "3", name: "Snowspeeder" }],
  },
];

export const heroeInfoWithNoStarshipsMock: FormattedFilms[] = [
  {
    title: "A New Hope",
    id: "1",
  },
  {
    title: "The Empire Strikes Back",
    id: "2",
  },
];

export const starshipsMock = [
  {
    MGLT: "75",
    cargo_capacity: "100000",
    consumables: "2 months",
    cost_in_credits: "3500000",
    created: "2014-12-12T11:05:37.420000Z",
    crew: "4",
    edited: "2014-12-22T17:35:44.431407Z",
    hyperdrive_rating: "1.0",
    length: "34.37",
    manufacturer: "Kuat Drive Yards",
    max_atmosphering_speed: "950",
    model: "Sentinel-class landing craft",
    name: "Sentinel-class landing craft",
    passengers: "75",
    films: [1, 2], // Example film IDs
    pilots: [],
    starship_class: "landing craft",
    url: "https://swapi.dev/api/starships/5/",
  },
  {
    MGLT: "10",
    cargo_capacity: "110",
    consumables: "1 week",
    cost_in_credits: "8000",
    created: "2014-12-15T12:31:42.547000Z",
    crew: "2",
    edited: "2014-12-20T21:23:49.897000Z",
    hyperdrive_rating: "1.0",
    length: "20",
    manufacturer: "Sienar Fleet Systems",
    max_atmosphering_speed: "1050",
    model: "TIE/LN starfighter",
    name: "TIE/LN starfighter",
    passengers: "0",
    films: [1], // Example film IDs
    pilots: [],
    starship_class: "Starfighter",
    url: "https://swapi.dev/api/starships/6/",
  },
];
