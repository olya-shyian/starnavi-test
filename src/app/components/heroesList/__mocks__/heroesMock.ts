import { IHeroe } from "@/app/interfaces/IHeroe";

export const HeroesMock: IHeroe[] = [
  {
    birth_year: "19BBY",
    eye_color: "Blue",
    films: [1, 2, 3, 4, 5, 6],
    gender: "Male",
    hair_color: "Blond",
    height: "172",
    homeworld: 1,
    mass: "77",
    name: "Luke Skywalker",
    skin_color: "Fair",
    created: "2022-01-01T00:00:00.000Z",
    edited: "2022-01-01T00:00:00.000Z",
    species: [1],
    starships: [1, 2, 3],
    url: "https://swapi.dev/api/people/1/",
    vehicles: [1, 2],
  },
  {
    birth_year: "112BBY",
    eye_color: "Yellow",
    films: [1, 2, 3],
    gender: "Female",
    hair_color: "Brown",
    height: "150",
    homeworld: 2,
    mass: "49",
    name: "Leia Organa",
    skin_color: "Light",
    created: "2022-01-01T00:00:00.000Z",
    edited: "2022-01-01T00:00:00.000Z",
    species: [1],
    starships: [3, 4],
    url: "https://swapi.dev/api/people/2/",
    vehicles: [3, 4],
  },
];
