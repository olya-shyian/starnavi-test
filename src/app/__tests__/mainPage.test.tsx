import React from "react";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import App from "../page";
import { HeroesMock } from "../components/heroesList/__mocks__/heroesMock";

const mockAxios = new MockAdapter(axios);

jest.mock("next/link", () => {
  const NextLink = ({ children }: { children: React.ReactNode }) => {
    return <>{children}</>;
  };

  NextLink.displayName = "NextLink";

  return NextLink;
});

jest.mock("next/navigation", () => ({
  useSearchParams: jest.fn(() => new URLSearchParams("")),
}));

describe("App Component", () => {
  beforeEach(() => {
    mockAxios.reset();
  });

  it("renders list of hero names on successful API call", async () => {
    // Mock the API call
    mockAxios.onGet("https://sw-api.starnavi.io/people?page=1").reply(200, {
      results: HeroesMock,
      count: HeroesMock.length,
    });

    await act(async () => {
      render(<App />);
    });

    // Verify the snapshot
    expect(screen.getByRole("main")).toMatchSnapshot();
  });
});
