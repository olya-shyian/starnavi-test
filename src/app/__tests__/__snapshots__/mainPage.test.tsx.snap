// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders list of hero names on successful API call 1`] = `
<main>
  <div
    class="css-h74a88"
  >
    <div
      class="css-1oqh2xe"
    >
      <h2
        class="chakra-heading css-wlea3r"
      >
        Oops! Something went wrong.
      </h2>
      <p
        class="chakra-text css-x4lmqn"
      >
        Unable to load todos
      </p>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Try Again
      </button>
    </div>
  </div>
</main>
`;
